### PR TITLE
[oneDNN v3.x]: Enabled reorder primitive cache and some misc. changes

### PR DESCRIPTION
**Pull request by bhavani-subramanian**
- This PR enables reorder primitive cache in `mkl_util.h` and `mkl_quantize_op.cc`.
- This PR passes all unit tests related to reorder primitive cache for oneDNN v3.x.
- This PR also enables oneDNN v3.x in benchmark tests.
- These changes are specific only to oneDNN kernels and do not affect Eigen ops.

### CODE REVIEW

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and sorry for the delay! I have one question.

File: tensorflow/core/util/mkl_util.h [hunk 1]
```diff
@@ -2157,64 +2136,99 @@ class MklReorderPrimitiveFactory : public MklPrimitiveFactory<T> {
...
+    int from_inner_nblks = from_desc.get_inner_nblks();
+    memory::dims from_inner_blks = from_desc.get_inner_blks();
+    memory::dims from_inner_idxs = from_desc.get_inner_idxs();
+
+    memory::dims to_dims = to_desc.get_dims();
+    int to_inner_nblks = to_desc.get_inner_nblks();
+    memory::dims to_inner_blks = to_desc.get_inner_blks();
+    memory::dims to_inner_idxs = to_desc.get_inner_idxs();
+    memory::dims to_strides = to_desc.get_strides();
+#endif  // !ENABLE_ONEDNN_V3
```
[the remark this answers is not in the document]
Comment (bhavani-subramanian) [reply tensorflow/core/util/mkl_util.h]: Makes sense. Fixed it.

**Comment by bhavani-subramanian** [comment]
> Could you please merge conflicts and rebase? All tests are failing and, for some reason, there are no logs. Please also reply to my comment on `mkl_util.h`. Thank you!

@penpornk  Thanks for reviewing this PR! I have addressed your review comments and also rebased with latest master. Please take a look.

File: tensorflow/core/kernels/mkl/mkl_fused_instance_norm_op.cc [hunk 2]
```diff
@@ -13,7 +13,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
-#ifdef INTEL_MKL
+#if defined(INTEL_MKL) && !defined(ENABLE_ONEDNN_V3)
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_fused_instance_norm_op.cc:16]: Could you please add `&& !defined(ENABLE_ONEDNN_V3)` in the comment at the `#endif` line at the end of the file as well?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the changes! One minor nit please.

File: tensorflow/core/kernels/mkl/mkl_fused_instance_norm_op.cc (hunk 2 above)
Comment (bhavani-subramanian) [reply tensorflow/core/kernels/mkl/mkl_fused_instance_norm_op.cc]: @penpornk Looks like this kernel got disabled for oneDNN v3.x after rebasing with master. I have fixed it now. Please take a look.

Comment (penpornk) [reply tensorflow/core/kernels/mkl/mkl_fused_instance_norm_op.cc]: Got it. Thank you!

**Review by penpornk — APPROVED** [review]

**Review by penpornk — APPROVED** [review]
Thank you for the sync!

**Pull request merged**